src/main/java/com/eternalcode/parcellockers/database/ParcelLockerDatabaseService.java

```diff
@@ -26,6 +26,7 @@
 public class ParcelLockerDatabaseService implements ParcelLockerRepository {
 
     private final Map<UUID, ParcelLocker> cache = new HashMap<>();
+    private final Map<Position, UUID> positionCache = new HashMap<>();
```

Comment:
I would create some methods for checking is someting already in map and creating etc.

Reply:
method for checking isn't really needed, i created modification methods though